src/States/BystanderInterventionState.js

```diff
@@ -0,0 +1,299 @@
+          console.log('c1');
+          break;
+
+        case 1:
+          key1 = 'choice2a';
+          key2 = 'choice2b';
+          console.log('c2');
+          break;
+
+        case 2:
+          key1 = 'choice3a';
+          key2 = 'choice3b';
+          console.log('c3');
+          break;
+
+        case 3:
+          key1 = 'choice4a';
+          key2 = 'choice4b';
+          console.log('c4');
+          break;
+
+        case 4:
+          key1 = 'choice5a';
+          key2 = 'choice5b';
+          console.log('c5');
+          break;
+
+        default:
+          console.log('error in switch');
+      }
+
+      choiceButton1 = createChoiceButton1(key1);
+      choiceButton1.inputEnabled = true;
+      choiceButton1.events.onInputUp.add(onChoiceSelected, {selected: 1});
+
+      choiceButton2 = createChoiceButton2(key2);
+      choiceButton2.inputEnabled = true;
+      choiceButton2.events.onInputUp.add(onChoiceSelected, {selected: 2});
+    }
+    else if (dialogueTree[progress].type == 'date') {
+      dateDialogue = createDateDialogue(dialogueTree[progress].msg);
+      setTimeout(displayNext, DIALOGUE_DISPLAY_TIME);
+      console.log('date');
+    }
+    else if (dialogueTree[progress].type == 'friend') {
+      friendDialogue = createFriendDialogue(dialogueTree[progress].msg);
+      setTimeout(displayNext, DIALOGUE_DISPLAY_TIME);
+      console.log('friend');
+    }
+    else {
+      console.log('unable to match type');
+    }
+
+    // Progress the dialogue tree
+    progress++;
+  }
+
+  /**
+   * onChoiceSelected: void -> void
+   * clears the screen of all buttons and/or dialogue, resets all to null, and then displays the next dialogue
+   */
+  function onChoiceSelected() {
+
+    // if (dateDialogue) {
+    //   dateDialogue.destroy();
+    // }
+    
+    // if (friendDialogue) {
+    //   friendDialogue.destroy();
+    // }
+
+    if (choiceButton1) {
+      choiceButton1.destroy();
+    }
+
+    if (choiceButton2) {
+      choiceButton2.destroy();
+    }
+
+    dateDialogue = null;
+    friendDialogue = null;
+    choiceButton1 = null;
+    choiceButton2 = null;
+    narrative = null;
+
+    // Display the next dialogue
+    displayNext();
+  }
+
+  /**
+   * Begin transition to the next state with a fade out.
+   */
+  function transitionToNextState() {
+    var properties = {alpha: 0};
+    var fadeOutDuration = 2000;
+    var ease = Phaser.Easing.Linear.None;
+    var autoStart = true;
+    var delay = 2000;
+    var repeat = false;
+    var yoyo = false;
+
+    game.add.tween(background).to(properties, fadeOutDuration, ease, autoStart, delay, repeat, yoyo);
+    setTimeout(startNextState, fadeOutDuration + delay);
+  }
+
+  /**
+   * Add and start next state. Starting a new state automatically shuts down the current one.
+   */
+  function startNextState() {
+    //game.state.add('at-prom', new AtProm());
+    //game.state.start('at-prom');
+
+    //@@@@@@@ INSERT END CREDITS HERE
```

Comment:
Could be worth getting into the habit of using `@todo` for todo-like comments. Gives us a standard string to search for and most IDEs can be setup to look for them and call them out.